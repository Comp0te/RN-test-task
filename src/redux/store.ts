import { createStore, applyMiddleware, combineReducers } from 'redux';
import { combineEpics, createEpicMiddleware } from 'redux-observable';
import { composeWithDevTools } from 'redux-devtools-extension';

import { requestsReducer } from './requests/reducers';

import { requestsEpics } from './requests/epics';

export const rootReducer = combineReducers({
  requests: requestsReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

const rootEpic = combineEpics(
  ...requestsEpics,
);

const epicMiddleware = createEpicMiddleware();

const middleware = [
  epicMiddleware,
];

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middleware)),
);

epicMiddleware.run(rootEpic);

export default store;
