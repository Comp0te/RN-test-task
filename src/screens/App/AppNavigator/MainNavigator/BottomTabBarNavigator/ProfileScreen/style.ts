import { StyleSheet } from 'react-native';

const style = StyleSheet.create({
  safeArea: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default style;