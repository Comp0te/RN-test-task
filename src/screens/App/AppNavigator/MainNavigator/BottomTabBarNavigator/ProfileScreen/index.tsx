import React, { useCallback } from 'react';
import { connect, DispatchProp } from 'react-redux';
import style from './style';

import { SafeAreaView } from 'react-native';
import CommonButton from '../../../../../../shared/components/CommonButton';

import * as authAC from '../../../../../../redux/auth/AC';

type Props = DispatchProp<authAC.Actions>;

const ProfileScreen: React.FC<Props> = (props) => {
  const {dispatch} = props;

  const onPressLogOut = useCallback(() => {
    dispatch(authAC.Actions.logout());
  }, []);

  return (
    <SafeAreaView style={style.safeArea}>
      <CommonButton
        title='SIGN OUT'
        onPress={onPressLogOut}
        iconName='sign-out'
      />
    </SafeAreaView>
  );
};

export default connect()(React.memo(ProfileScreen));
