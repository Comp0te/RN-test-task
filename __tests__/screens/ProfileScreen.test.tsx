import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProfileScreen from '../../src/screens/App/AppNavigator/MainNavigator/BottomTabBarNavigator/ProfileScreen';
import { Provider } from 'react-redux';
import store from '../../src/redux/store';

it('ProfileScreen renders correctly', () => {
  expect(renderer.create(
    <Provider store={store}>
      <ProfileScreen/>
    </Provider>,
  )).toMatchSnapshot();
});