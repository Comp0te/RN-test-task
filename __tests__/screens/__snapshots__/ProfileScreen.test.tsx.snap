// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#860483",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 20,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              ï‚‹
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          SIGN OUT
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
